backend/controllers/followups.ts

```diff
@@ -42,29 +43,40 @@ export function resultRedirect(req: ajRequest, res: Response) {
   res.redirect(req.simulation.returnPath)
 }
 
-export function persist(req: ajRequest, res: Response) {
+export async function persist(req: ajRequest, res: Response) {
   if (!req.body.email || !req.body.email.length) {
     return res.status(400).send({ result: "KO" })
   }
 
-  Followup.create({
-    simulation: req.simulation,
-    email: req.body.email,
-    surveyOptin: req.body.surveyOptin,
-  })
-    .then((followup) => {
-      req.simulation.hasFollowup = true
-      return req.simulation
-        .save()
-        .then(() => followup.sendSimulationResultsEmail())
-    })
-    .then(() => {
-      return res.send({ result: "OK" })
-    })
-    .catch((error) => {
-      console.error("error", error)
-      return res.status(400).send({ result: "KO" })
+  const simulation = req.simulation
```

Comment:
@guillett la **proposition** dont je te parlais c'est de sortir a logique de creation d'un followup valide dans le controller (tbh ce premier point est ok j'aimerais le sortir dans un service qui sera utilisé ici).

L'objectif est double : 
- D'une part c'est d'avoir tout au même endroit, pas de logique cachée dans le pre-save (ou encore pire dans le render de l'email). Le pre-save c'est un peu caché et j'imagine un futur dev où on se demande d'où vient de `version` sans penser au fait qu'il y a un presave
- D'autre part (mais à valider avec la team) c'est de ce dire que le model mongoose défini ce qu'est un followup valide, mais pas forcément comment on en fait un (separation of concern)

cc @Shamzic @Valandr @Cugniere

Reply:
Je ne trouve pas idéal d'avoir utilisé le _pattern factory_ sans empêcher la création d'un follow-up via _create_ en dehors de la factory.

L'extraction dans une fonction dédiée, incluse dans le fichier du modèle et utilisée dans le _pre_ faute de mieux me paraissait plus propre mais bon c'est peut-être qu'une histoire de goût.

Reply:
Ah mince j'avais déjà merge, mais la discussion peut continuer :-) 

- Le mot factory est p.e. pas super bien choisi, car c'est pas a proprement parlé un pattern factory il me semble, je peux changer pour "creator" pour moins de confusion ? 
- En lisant ta remarque, j'ai l'impression que c'est justement ma remarque/question précédente où je propose de séparer la logique de validation (dans le model) de la logique de creation (là on peut utiliser le service, mais aussi le faire manuellement, la validation mongoose DOIT nous raise une erreur si on ne fait ça pas bien).

J'ai l'impression que ta solution cache la complexité et a un coté inflexible (1). Ça peut poser soucis par exemple le jours où on doit importer des followup via un script ou même tester correctement la couche de validation. 

(1) en bdd il le faut l'inflexibilité mais la validation de schema reste là pour s'en assurer.